### PR TITLE
liblapin: init at 0-unstable-2024-05-20

### PR DESCRIPTION
**Pull request by Sigmanificient**
## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Sigmanificient** [comment]
note: II am not sure on what license to put on it, see [the license file](https://github.com/Damdoshi/LibLapin/blob/master/license.txt)

**Comment by Sigmanificient** [comment]
note: see the following issue for the `src/camera` build error with `opencv`:
- https://github.com/NixOS/nixpkgs/issues/313062

**Review by OPNA2608 — CHANGES_REQUESTED** [review]

File: pkgs/by-name/li/liblapin/package.nix [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  ];
+
+  outputs = [ "out" "dev" ];
+
+  meta = {
+    description = "Multimedia library for rookies and prototyping";
+    homepage = "http://liblapin.org";
+    platforms = lib.platforms.unix;
+    maintainers = with lib.maintainers; [ sigmanificient ];
+  };
```
[the remark this answers is not in the document]
Comment (Sigmanificient) [reply pkgs/by-name/li/liblapin/package.nix:56]: See the note about that

[the remark this answers is not in the document]
Comment (OPNA2608) [reply pkgs/by-name/li/liblapin/package.nix:56]: Ah sorry. License seems to be [3-clause BSD](https://opensource.org/license/BSD-3-clause).

File: pkgs/by-name/li/liblapin/package.nix [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  prePatch = ''
+    # camera module fails to build with opencv, due to missing V4L2 support
+    rm -rf src/camera
+
+    substituteInPlace Makefile \
+      --replace-fail "/bin/echo" "echo"
+  '';
+
+  enableParallelBuilding = true;
+  hardeningDisable = [ "fortify" ];
```
[the remark this answers is not in the document]
Comment (Sigmanificient) [reply pkgs/by-name/li/liblapin/package.nix:31]: not necessarily, but it avoid extra warning from nix hardening profile, de to `write` calls return value being unused.

**Comment by OPNA2608** [comment]
Could you also fix the version in the commit title please?

**Comment by Sigmanificient** [comment]
> Could you also fix the version in the commit title please?

Done!

**Review by SuperSandro2000 — APPROVED** [review]

**Review by OPNA2608 — APPROVED** [review]
LGTM.

**Pull request merged**